tools/c7n_azure/c7n_azure/utils.py

```diff
@@ -380,6 +381,29 @@ def build_ports_dict(nsg, direction_key, ip_protocol):
         return ports
 
 
+class IpRangeHelper(object):
+
+    @staticmethod
+    def parse_ip_ranges(dict, key):
+        '''
+        Parses IP range or CIDR mask.
```

Comment:
Does it work with a single IP?

Reply:
It must be either two-part range or a single network in the CIDR format. For a single IP you do 0.0.0.0-0.0.0.0 consistently with the portal UI.

Reply:
This is a helper method that can be reused for Azure Firewall (Storage, KeyVault etc), so I think common code needs to have a support for all 3 methods: range, cidr, single ip

Reply:
I checked single IP it works :) it assumes /32 subnet.